src/k8s/pkg/component/hook.go

```diff
@@ -0,0 +1,93 @@
+package component
+
+import (
+	"fmt"
+	"net"
+	"strings"
+
+	"github.com/canonical/k8s/pkg/k8s/utils"
+	"github.com/canonical/k8s/pkg/snap"
+)
+
+type valuesHook func() (map[string]interface{}, error)
+
+var valuesHooks = map[string]valuesHook{
+	"network": networkValues,
+}
+
+func networkValues() (map[string]interface{}, error) {
```

Comment:
here and in a lot of other places: Use `any` instead of `interface{}`

https://google.github.io/styleguide/go/decisions#use-any